pkg/kubelet/types.go

```diff
@@ -71,3 +73,20 @@ type PodUpdate struct {
 func GetPodFullName(pod *api.BoundPod) string {
 	return fmt.Sprintf("%s.%s.%s", pod.Name, pod.Namespace, pod.Annotations[ConfigSourceAnnotationKey])
 }
+
+// ParsePodFullName unpacks a pod full name and returns the pod name, namespace, and annotations.
+// If the pod full name is invalid, empty strings are returend.
+func ParsePodFullName(podFullName string) (podName, podNamespace string, podAnnotations map[string]string) {
+	parts := strings.Split(podFullName, ".")
+	expectedNumFields := 3
+	actualNumFields := len(parts)
+	if actualNumFields != expectedNumFields {
+		glog.Warningf("found a podFullName (%q) with too few fields: expected %d, actual %d.", podFullName, expectedNumFields, actualNumFields)
+		return
+	}
+	podName = parts[0]
+	podNamespace = parts[1]
+	podAnnotations = make(map[string]string)
+	podAnnotations[ConfigSourceAnnotationKey] = parts[2]
```

Comment:
do we really want to document the third field generic annotations, or is it specifically the config source?  I.e. will there ever be a second annotation in here, and if so, what does that look like (since all we have is a value today, and not a key+value) ?